test/conformance/queue/urQueueFinish.cpp

```diff
@@ -0,0 +1,25 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: MIT
+#include <uur/fixtures.h>
+
+using urQueueFinishTest = uur::urQueueTest;
+UUR_INSTANTIATE_DEVICE_TEST_SUITE_P(urQueueFinishTest);
+
+TEST_P(urQueueFinishTest, Success) {
+    constexpr size_t buffer_size = 1024;
+    ur_mem_handle_t buffer = nullptr;
+    ASSERT_SUCCESS(urMemBufferCreate(context, UR_MEM_FLAG_READ_WRITE,
+                                     buffer_size, nullptr, &buffer));
+
+    std::vector<uint8_t> data(buffer_size, 42);
+    ASSERT_SUCCESS(urEnqueueMemBufferWrite(queue, buffer, /* blocking */ false,
+                                           0, 1024, data.data(), 0, nullptr,
+                                           nullptr));
+
+    ASSERT_SUCCESS(urQueueFinish(queue));
```

Comment:
Should we validate that the enqueued tasks have actually finished after calling urQueueFinish? The function could return success but actually do nothing.

Reply:
Finish will block until all commands to complete.

Reply:
Ah sorry I misunderstood your comment. Yes you're right - I'll add a check. I assume you can just query the status of the event right?

Reply:
yeah, I think that would be a good way to do it :)

Reply:
done.